src/plugins/definition/schema.js

```diff
@@ -0,0 +1,203 @@
+// Copyright 2019 OpenStax Poland
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license text.
+
+import { Text, Block } from 'slate'
+
+function normalizeDefinition(change, error) {
+    const { code, key, index, node, child } = error
+
+    switch (code) {
+    case 'first_child_type_invalid':
+        // Insert missing definition_term
+        const term = Block.create({
+            type: 'definition_term',
+            nodes: [Text.create()],
+        })
+        change.insertNodeByKey(node.key, 0, term)
+        break
+
+    case 'child_type_invalid':
+        if (Text.isText(child)) {
+            change.setNodeByKey(child.key, 'paragraph')
+            break
+        }
+        if (child.type === 'paragraph') {
+            change.wrapNodeByKey(child.key, Block.create('definition_meaning'))
+            break
+        }
+
+        console.warn('Unhandled definition child_type_invalid:', child)
+        break
+
+    case 'child_min_invalid':
+        if (!node.nodes.some(n => n.type && n.type === 'definition_meaning')) {
+            // Insert missing meaning.
+            change.insertMeaning()
+            break
+        }
+
+        console.log('Unhandled definition child_min_invalid:', child)
+        break
+
+    case 'child_unknown':
+        // An invalid node following the See Also should be folded into the
+        // See Also.
+        if (index > 0) {
+            const nodeBefore = node.nodes.get(index - 1)
+            if (nodeBefore && nodeBefore.type === 'definition_seealso') {
+                change.moveNodeByKey(child.key, nodeBefore.key, nodeBefore.nodes.size)
+                break
+            }
+        }
+
+        // Unwrap any other nodes
+        change.unwrapBlockByKey(child.key)
+        break
+
+    default:
+        console.warn('Unhandled definition violation:', code)
+        break
+    }
+}
+
+function normalizeTerm(change, error) {
+    const { code, key, node, child } = error
+
+    switch (code) {
+    case 'parent_type_invalid':
+        change.unwrapBlockByKey(node.key)
+        break
```

Comment:
This will insert raw text nodes into blocks which only accept blocks as children.